src/filelock/_api.py

```diff
@@ -73,15 +75,36 @@ class ThreadLocalFileContext(FileLockContext, local):
     """A thread local version of the ``FileLockContext`` class."""
 
 
-class BaseFileLock(ABC, contextlib.ContextDecorator):
+class SingletonPerLockFileMeta(ABCMeta):
+    """
+    Metaclass for the ``BaseFileLock`` class which ensures that only one instance of the class is created per
+    lock (when specified).
+    """
+
+    _instances: ClassVar[WeakValueDictionary] = WeakValueDictionary()
+
+    def __call__(cls, lock_file: str | os.PathLike[str], *args: Sequence, **kwargs: Mapping) -> object:
```

Comment:
Hmm just realizing the same behavior can be achieved via `__new__` instead of creating a metaclass. I'm going to refactor the approach since that seems clearer.